gen/gen.go

```diff
@@ -0,0 +1,154 @@
+		log.Fatalln(err)
+	}
+}
+
+type IntegrationResource struct {
+	ResourceClassName     string
+	TerraformResourceName string
+}
+
+// generateIntegrationResources generates Terraform resources for Mondoo's integrations.
+func generateIntegrationResources() error {
+	// Read the template file
+	resourceTemplateFile := filepath.Join("gen", "templates", "integration_resource.go.tmpl")
+	resourceTmpl, err := template.ParseFiles(resourceTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	testTemplateFile := filepath.Join("gen", "templates", "integration_resource_test.go.tmpl")
+	testTmpl, err := template.ParseFiles(testTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	resourceDotTFTemplateFile := filepath.Join("gen", "templates", "resource.tf.tmpl")
+	resourceTFTmpl, err := template.ParseFiles(resourceDotTFTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	// Ensure the output directory exists
+	outputDirPath := filepath.Join("gen", "generated")
+	if err := os.MkdirAll(outputDirPath, 0755); err != nil {
+		return err
+	}
+
+	i := mondoov1.ClientIntegrationConfigurationInput{}
+	m := structs.Map(i)
+
+	for k := range m {
+		// TODO we know the type and the struct associated to the type, we need
+		// to look it (the struct) and use the same `structs.Map(v)` to list all
+		// fields per integration and auto generate the terraform schema and more
+		// details, for now, we only leave a comment where we need to add specific
+		// integration options
+
+		var (
+			className, _          = strings.CutSuffix(k, "ConfigurationOptions")
+			terraformResourceName = strings.ToLower(toSnakeCase(className))
+			resource              = IntegrationResource{
+				ResourceClassName:     className,
+				TerraformResourceName: terraformResourceName,
+			}
+		)
+
+		fmt.Printf("> Generating code for %s integration\n", className)
+
+		// Create the resource file
+		resourceOutputFilePath := filepath.Join(outputDirPath,
+			fmt.Sprintf("integration_%s_resource.go", terraformResourceName),
+		)
+		resourceFile, err := os.Create(resourceOutputFilePath)
+		if err != nil {
+			return err
+		}
+		defer resourceFile.Close()
+
+		if err := resourceTmpl.Execute(resourceFile, resource); err != nil {
+			return err
+		}
+
+		// Create test file
+		testOutputFilePath := filepath.Join(outputDirPath,
+			fmt.Sprintf("integration_%s_resource_test.go", terraformResourceName),
+		)
+		testFile, err := os.Create(testOutputFilePath)
+		if err != nil {
+			return err
+		}
+		defer testFile.Close()
+
+		if err := testTmpl.Execute(testFile, resource); err != nil {
+			return err
+		}
+
+		// Create examples/ files
+		examplesDirPath := filepath.Join(outputDirPath, "examples", fmt.Sprintf("mondoo_integration_%s", terraformResourceName))
+		// Ensure the output directory exists
+		if err := os.MkdirAll(examplesDirPath, 0755); err != nil {
+			return err
+		}
+		// Create example main.tf
+		err = os.WriteFile(filepath.Join(examplesDirPath, "main.tf"), mainDotTFTestFile(), 0644)
```

Comment:
Same here: Could we generate this directly in the final directory?

Reply:
not yet. still need some more work but we will ultimately write it in the right directory.

Reply:
Is this something for a follow-up or do you want to include it in this PR?

Reply:
No longer a follow-up. This PR now creates two auto-generated integrations.

<img src="https://media2.giphy.com/media/v1.Y2lkPWJkM2VhNTdlbzBseGd5bDlyOWpvb3B0d2o3aHFnajc0MWE5enB1M2l1YTRnNnUxaiZlcD12MV9naWZzX3NlYXJjaCZjdD1n/8wfoaIjVc0FBaLu5xH/giphy.gif"/>